### PR TITLE
Biome layer system mapping

### PR DESCRIPTION
**Pull request by Gegy**
Start to mapping the key components of the biome layer system

### CODE REVIEW

File: mappings/net/minecraft/util/math/layer/InitLayer.mapping [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CLASS cfo net/minecraft/util/math/layer/InitLayer
```
Comment (kashike) [inline mappings/net/minecraft/util/math/layer/InitLayer.mapping]: These should be in `net/minecraft/world/biome/layer/`.

**Review by kashike — CHANGES_REQUESTED** [review]

**Comment by asiekierka** [comment]
Uh, could you look at my [WIP] work in another PR? Maybe you can build on that work.

File: mappings/net/minecraft/util/math/layer/InitLayer.mapping (hunk 1 above)
Comment (Gegy) [reply mappings/net/minecraft/util/math/layer/InitLayer.mapping]: I did consider this, though `InitLayer` is a base interface for any layer that does not have a parent, so it isn't really tied to the biome system.

**Comment by Gegy** [comment]
Shall I integrate those mappings into this PR?

File: mappings/net/minecraft/util/math/layer/InitLayer.mapping (hunk 1 above)
Comment (kashike) [reply mappings/net/minecraft/util/math/layer/InitLayer.mapping]: The entire system is for biome layers, there's zero point having this separate.

**Comment by asiekierka** [comment]
Not really; they're not great, but you can use them as a base as I have figured some related-but-indirect things out.

Also, just to make sure; this is your own work, right?

**Comment by Gegy** [comment]
Alright.
I have not explored MCP's 1.13 mappings at all, so I can assure that I have not been influenced by that. x)

**Comment by asiekierka** [comment]
... Or OML?

**Comment by Gegy** [comment]
I partially mapped this system (albeit badly) for OML in 1.13. It has changed a fair bit since then, and I have not worked with those mappings in a while. Can be compared to [OML pomf](https://github.com/OpenModLoader/pomf/tree/18w43c/mappings/net/minecraft/util/math/layer) if that is needed, but these have turned out fairly different.

**Comment by asiekierka** [comment]
Okay. I'm just worried about potential taint issues from OML, but this is fine. I'll tentatively call @kashike in just to make sure, though.

**Comment by Gegy** [comment]
Alright, have moved layer package and believe I have integrated everything from asie's PR that this was missing. :)

File: mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS cez net/minecraft/world/biome/layer/BiomeLayers
+CLASS cez net/minecraft/world/biome/layer/VanillaBiomeLayers
```
Comment (kashike) [inline mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: Why? You don't see us doing `VanillaBlocks`, etc.

Comment (Gegy) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: From https://github.com/FabricMC/yarn/pull/248/files#diff-db683513994329b7b6b9b89e7d6a283bR1

Comment (kashike) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: @asiekierka I disagree with that change.

Comment (asiekierka) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: @gegy1000 The names were garbage and mostly meant for understanding; I'd rather you use them as inspiration and not directly copy them in.

Do BiomeLayers... I don't remember why I did that anymore

Comment (Gegy) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: The class constructs the layer stack for the overworld -- possibly `OverworldLayerBuilder` or something similar?

Comment (asiekierka) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: Yes, that's what i was going for - I wanted to emphasize it's specific to the overworld, not other dimensions... but maybe that's silly

Comment (Gegy) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: Went for `OverworldBiomeLayers` as `OverworldBiomeLayerBuilder` might be too long.

Comment (kashike) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: Except it isn't just used for constructing the layers for the overworld, it also has generic utilities. `BiomeLayers` is what it should be.

Comment (Gegy) [reply mappings/net/minecraft/world/biome/layer/VanillaBiomeLayers.mapping]: The only utilities it has are specifically relating to overworld biomes, and only used as such. Though I don't have a problem with changing it, `BiomeLayers` is still an effective name.

**Review by asiekierka — APPROVED** [review]

**Review by SquidDev — APPROVED** [review]

**Review by kashike — APPROVED** [review]
Good enough for a first pass.

**Review by modmuss50 — APPROVED** [review]

**Comment by Gegy** [comment]
Resolved merge conflicts

**Comment by modmuss50** [comment]
Thanks, for merging it. I was in the middle of doing it 👍

**Pull request merged**